src/packages/engine-core/src/NAPIEngine.ts

```diff
@@ -158,17 +159,19 @@ You may have to run ${chalk.greenBright(
   private async loadEngine(): Promise<void> {
     debug('loadEngine')
     if (!this.engine) {
-      if (!this.QueryEngine) {
+      if (!this.QueryEngineConstructor) {
         if (!this.libQueryEnginePath) {
+          // TODO Why is this executed twice
           this.libQueryEnginePath = await this.getLibQueryEnginePath()
           debug(`using ${this.libQueryEnginePath}`)
         }
         try {
           // this require needs to be resolved at runtime, tell webpack to ignore it
           this.library = eval('require')(this.libQueryEnginePath) as NAPI
-          this.QueryEngine = this.library.QueryEngine
+          this.QueryEngineConstructor = this.library.QueryEngine
         } catch (e) {
           if (fs.existsSync(this.libQueryEnginePath)) {
+            // TODO Check if this is tested
```

Comment:
Random thought: 

This could be related to the failure in @Jolg42's PR tests, where when later trying to `require` the library it fails. Maybe we can already do that here? Can we "test require" without actually requiring already?

https://github.com/prisma/prisma/runs/2849230129#step:7:713

Reply:
Unsure what you mean. The error above is todo with the Node-API `engineCommands` which lives in the `sdk`

Reply:
Yes, but those are executed _later_ in the process than this, right? So if we would already check here if we can successfully require whatever file we downloaded, this could avoid the later failure.